### PR TITLE
Feature/transaction notifications

### PR DESCRIPTION
**Pull request by Amirjab21**

### CODE REVIEW

File: packages/app/components/Notifications/ToastContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { bigNumberToNumber } from "@popcorn/utils";
+import toast from "react-hot-toast";
+
+export enum methodNames {
```
Comment (RedVeil) [inline packages/app/components/Notifications/ToastContainer.tsx:4]: I think this should be more explicit. Maybe `ToastMethod`.

File: packages/app/components/Notifications/ToastContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  return toast[messageType](title, options);
+}
+
+const customNotification = (methodName: methodNames, res: any, messageType: messageType, title: string, args: any) => {
+  switch (methodName) {
+    case methodNames.ZapIntoBatch:
+      const eventLog = res.events.filter((event) => event.event === "ZappedIntoBatch").args;
+      return toast[messageType](
```
Comment (RedVeil) [inline packages/app/components/Notifications/ToastContainer.tsx:47]: I think we have to use toast.custom() to display a toast that fully looks like the design.

Comment (RedVeil) [reply packages/app/components/Notifications/ToastContainer.tsx:47]: (With changed border radius etc. You can find docs for it on their site. It works perfect with tailwind)

File: packages/app/components/Notifications/ToastContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { bigNumberToNumber } from "@popcorn/utils";
+import toast from "react-hot-toast";
+
+export enum methodNames {
+  ZapIntoBatch = "ZapIntoBatch",
+}
+
+export enum messageType {
```
Comment (RedVeil) [inline packages/app/components/Notifications/ToastContainer.tsx:8]: I think we should make these also `ToastMethods`. We have to customize all calls now anyways so we can also spend the time and go into each function call to set these things explicitly

File: packages/app/components/Notifications/ToastContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  success = "success",
+  error = "error",
+  loading = "loading",
+  transactionSuccess = "success",
+}
+
+const options = {};
+
+export default function ToastContainer(
```
Comment (RedVeil) [inline packages/app/components/Notifications/ToastContainer.tsx:17]: Id like to rename this to `showToast` or smth like that. Its a bit easier to understand in the functions that call it. The original syntax is great `toast.success()`! Id like to get close to smth like that.

File: packages/app/components/Notifications/ToastContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+export enum messageType {
+  success = "success",
+  error = "error",
+  loading = "loading",
+  transactionSuccess = "success",
+}
+
+const options = {};
```
Comment (RedVeil) [inline packages/app/components/Notifications/ToastContainer.tsx:15]: What are options for?

File: packages/app/components/Notifications/ToastContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  messageType: messageType,
+  res?: any,
+  methodName?: methodNames,
+  args?: any,
+  description?: string,
+) {
+  toast.dismiss();
+  if (methodName) {
+    return customNotification(methodName, res, messageType, title, args);
+  } else if (description) {
```
Comment (RedVeil) [inline packages/app/components/Notifications/ToastContainer.tsx:28]: When we use only `MethodType` we should have a base toast container which just swaps out the text and icon via a switch case and a customNotification as you did here which again uses a switch case for more specialized designs

File: packages/app/pages/[network]/butter/index.tsx [hunk 7]
```diff
@@ -238,31 +238,41 @@ export default function Butter(): JSX.Element {
...
-        },
-        "hideMintPopover",
-        dispatch,
-      );
-    });
+          "hideMintPopover",
+          dispatch,
+        );
+      },
+      methodNames.ZapIntoBatch,
```
Comment (RedVeil) [inline packages/app/pages/[network]/butter/index.tsx:269]: This is not zapping into the batch but only depositing into the current batch

File: packages/app/components/Notifications/ToastContainer.tsx (hunk 1 above)
Comment (RedVeil) [reply packages/app/components/Notifications/ToastContainer.tsx:4]: We would need to decide maybe with @FanFanDAO and @anthonymartin how many different methods we want to have.

File: packages/app/components/Notifications/ToastContainer.tsx (hunk 5 above)
Comment (Amirjab21) [reply packages/app/components/Notifications/ToastContainer.tsx:15]: removed

File: packages/app/components/Notifications/ToastContainer.tsx (hunk 2 above)
Comment (Amirjab21) [reply packages/app/components/Notifications/ToastContainer.tsx:47]: Theres a problem with the toast.custom() - the animation just doesn't work properly.

Look at this issue: https://github.com/timolins/react-hot-toast/issues/116

I've put it on the latest PR but it doesn't work as others have also complained about.

File: packages/app/pages/[network]/butter/index.tsx (hunk 7 above)
Comment (Amirjab21) [reply packages/app/pages/[network]/butter/index.tsx]: done

File: packages/app/components/Notifications/ToastContainer.tsx (hunk 6 above)
Comment (Amirjab21) [reply packages/app/components/Notifications/ToastContainer.tsx]: Think I've implemented it - check it out

**Comment by RedVeil** [comment]
@Amirjab21 Great PR! Thinking more about it though i think we should get rid of custom notifications. All notifications should be custom either of type loading, success or error. Error works perfectly as is but we should simply input a <p> with the custom information in loading and success.

**Comment by RedVeil** [comment]
@Amirjab21 Additionally i think its time that onContractSuccess gets an input object with all the params. its getting kinda hard to understand when adding input params what is required and what they actually are there for.

**Comment by RedVeil** [comment]
@Amirjab21 Also can we rename ShowToast to TransactionToast?

**Comment by RedVeil** [comment]
i transfered this code into a new PR since it was easier than merging this into new-website

**Pull request closed without merging**